core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -408,9 +428,12 @@ abstract class AbstractFetcherThread(name: String,
   def markPartitionsForTruncation(topicPartition: TopicPartition, truncationOffset: Long): Unit = {
     partitionMapLock.lockInterruptibly()
     try {
+      // It is safe to reset `lastFetchedEpoch` here since we don't expect diverging offsets
```

Comment:
Can be done separately, but it would be nice to figure out how to move this logic into `ReplicaAlterLogDirManager` since this comment seems to only make sense if we assume this is the log dir fetcher and reconciliation with the leader has already completed. 

In fact, I wonder if it is possible to get rid of this code entirely. If the log dir fetcher is also tracking `lastFetchedEpoch`, then we could rely on detecting truncation dynamically through `ReplicaManager.fetchMessages` instead of the current somewhat clumsy coordination with the replica fetcher.

Reply:
Updated. The method is still there for older versions, but it is now disabled with IBP 2.7.